testsuite/classlibrary/TestLinkClock.sc

```diff
@@ -202,4 +202,140 @@ TestLinkClock : UnitTest {
 
 		semaphore.wait
 	}
+
+	test_LinkClock_meter_change_beats_keep_continuity {
+		var clock1, clock2,
+		tempo = 12, beatDurThresh = 0.4 / tempo,
+		secList1 = List.new, secList2 = List.new,
+		cond = Condition({ secList1.size == 20 and: { secList2.size == 20 } });
+
+		clock1 = LinkClock(tempo);
+		{
+			2.5.wait;  // offset clock starts
+			clock2 = LinkClock(tempo);
+			1.5.wait;
+			{
+				20.do {
+					secList2.add(clock2.seconds);
+					1.0.wait;
+				};
+				cond.signal;
+			}.fork(clock2);
+			20.do {
+				secList1.add(clock1.seconds);
+				clock1.setMeterAtBeat(rrand(2, 4), clock1.beats);
+				1.0.wait;
+			};
+			cond.signal;
+		}.fork(clock1);
+		cond.wait;
+		this.ifAsserts(
+			clock1.numPeers == 1,
+			"no extra clocks in session",
+			{
+				this.assert(
+					(secList1 absdif: secList2).every(_ <= beatDurThresh),
+					"LinkClock:setMeterAtBeat should not disrupt beat duration\nclock1: %\nclock2: %\n"
+					.format(secList1, secList2)
+				)
+			},
+			{
+				this.assert(false, "Make sure no other LinkClocks are running while testing");
+			}
+		);
+		clock1.stop;
+		clock2.stop;
+	}
+
+	// this test uses "magic number" wait statements
+	// because it takes a little time for a new LinkClock to stabilize
+	// and we have no signal to know when it's ready
+	test_LinkClock_sync_meter_aligns_barlines {
```

Comment:
this test causes extra printing when run: "syncing meter to 3, base = 2.0". can you find a way to avoid that please?